array.c

```diff
@@ -4386,6 +4386,27 @@ rb_ary_union_multi(int argc, VALUE *argv, VALUE ary)
     return ary_union;
 }
 
+/*
+ *  call-seq:
+ *     ary.intersect?(other_ary)   -> true or false
+ *
+ *  Returns +true+ if the array and +other_ary+ have at least one element in
+ *  common, otherwise returns +false+.
+ *
+ *     a = [ 1, 2, 3 ]
+ *     b = [ 3, 4, 5 ]
+ *     c = [ 5, 6, 7 ]
+ *     a.intersect?(b)   #=> true
+ *     a.intersect?(c)   #=> false
+ */
+
+static VALUE
+rb_ary_intersects(VALUE ary1, VALUE ary2)
+{
+    VALUE intersect = rb_ary_and(ary1, ary2);
```

Comment:
This creates an intermediate array.

Reply:
Is that a problem? I could dup some of the code from `rb_ary_and` to prevent that if that's preferable.

Reply:
@nobu I pulled some code out of `rb_ary_and` and tweaked it. It should be a lot more efficient now.